### PR TITLE
docs(readme): update and split readme to readme and contributing

### PR DESCRIPTION
**Pull request by agankarin**

### CODE REVIEW

**Comment by tomenden** [comment]
Tests are failing, I guess we need to remove the snapshot tests and replace them with better ones before updating corvid-types, right?
I think I'll put someone on it next week

**Comment by agankarin** [comment]
The JSON model os broken(has duplicate declaration and a missing declaration).
Master is also broken :( (Sam is on it)
Anyway, this PR just updates the README, can you please review it? :)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Build
+The declaration files are auto generated(as there are new corvid APIs on a monthly base)
```
Comment (tomenden) [inline CONTRIBUTING.md]: Grammer - Monthly basis.
Perhaps not even monthly though, I would say (as new apis are added all the time, or something like that. Feel free to consult sam about the english)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+The declaration files are auto generated(as there are new corvid APIs on a monthly base) 
+
+To generate the files, simply run
+`npm i && npm run build`
+
+# Test
+`npm run test`
+
+# Publish to npm
+On master branch, run [`npm version [<newversion> | major | minor | patch | ...] -m <commit message>`](https://docs.npmjs.com/cli/version)
```
Comment (tomenden) [inline CONTRIBUTING.md]: Why is it different here from corvid-cli in which we don't do it directly on master? 🤔

File: README.md [hunk 3]
```diff
@@ -1,13 +1,38 @@
...
+<p align="center">
+  <b>TypeScript type definitions for <a href="https://www.wix.com/corvid/reference/">Corvid APIs</a></b>
+</p>
+<br>
+
+[![npm version](https://img.shields.io/npm/v/corvid-types.svg?style=flat)](https://www.npmjs.com/package/corvid-types)
+[![Status](http://img.shields.io/travis/com/wix-incubator/corvid-types/master.svg?style=flat)](https://travis-ci.com/wix-incubator/corvid-types)  
+
+> **Important:**
+Features relating to the Local Editor are currently in the stage of alpha release and are still being developed. Future functionality may not be backward compatible. When using these features, you assume the risks associated with the use of such an alpha version.
```
Comment (tomenden) [inline README.md]: Is this copy-paste from the docs? If not, consult Sam

File: README.md [hunk 4]
```diff
@@ -1,13 +1,38 @@
...
 
-# Installation
-> `npm install --save-dev corvid-types`
 
-# Test
-`npm run test`
+This package contains type definitions for Corvid by Wix.
+Intended to be used with [`corvid-cli`](https://github.com/wix-incubator/corvid#corvid-local-development) to provide stunning code intelligence experiment with your local IDE.
```
Comment (tomenden) [inline README.md]: Experiment -> experience

File: README.md [hunk 5]
```diff
@@ -1,13 +1,38 @@
...
-# Installation
-> `npm install --save-dev corvid-types`
 
-# Test
-`npm run test`
+This package contains type definitions for Corvid by Wix.
+Intended to be used with [`corvid-cli`](https://github.com/wix-incubator/corvid#corvid-local-development) to provide stunning code intelligence experiment with your local IDE.    
+
+# How to use
+You can install the `corvid-types` package manually or let [`create-corvid-app`](https://www.npmjs.com/package/create-corvid-app) done everything for you!
```
Comment (tomenden) [inline README.md]: Done -> do

File: README.md (hunk 3 above)
Comment (agankarin) [reply README.md]: yep, just a copy-past

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# Build
+The declaration files are auto generated(as new Corvid apis are added all the time)
```
Comment (wixysam) [inline CONTRIBUTING.md]: - Use `autogenerated` or `auto-generated`.
- Change to Corvid APIs
- You can remove the parenthesis and add a period.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+# Publish to npm
+On master branch, run [`npm version [<newversion> | major | minor | patch | ...] -m <commit message>`](https://docs.npmjs.com/cli/version)
+
+The `npm version` command performs the following tasks:
+1. Update the version in package.json to the new version
+2. Create a new commit
+3. Create a git tag
+
+Next, just push you changes to github including your new tag by running `git push origin master --tag`
```
Comment (wixysam) [inline CONTRIBUTING.md]: - `GitHub, including you new tag, by...`
- Move command to the next line.

File: README.md [hunk 8]
```diff
@@ -1,13 +1,38 @@
...
+> **Important:**
+Features relating to the Local Editor are currently in the stage of alpha release and are still being developed. Future functionality may not be backward compatible. When using these features, you assume the risks associated with the use of such an alpha version.
 
-# Installation
-> `npm install --save-dev corvid-types`
 
-# Test
-`npm run test`
+This package contains type definitions for Corvid by Wix.
+Intended to be used with [`corvid-cli`](https://github.com/wix-incubator/corvid#corvid-local-development) to provide stunning code intelligence experience with your local IDE.
```
Comment (wixysam) [inline README.md]: `It is intended to be used with the...to provide a...in your local IDE.`

File: README.md [hunk 9]
```diff
@@ -1,13 +1,38 @@
...
 
-# Installation
-> `npm install --save-dev corvid-types`
 
-# Test
-`npm run test`
+This package contains type definitions for Corvid by Wix.
+Intended to be used with [`corvid-cli`](https://github.com/wix-incubator/corvid#corvid-local-development) to provide stunning code intelligence experience with your local IDE.    
+
+# How to use
```
Comment (wixysam) [inline README.md]: How to Use

File: README.md [hunk 10]
```diff
@@ -1,13 +1,38 @@
...
+You can install the `corvid-types` package manually or let [`create-corvid-app`](https://www.npmjs.com/package/create-corvid-app) do everything for you!
+### create-corvid-app
+Just run 
+    ```
+    npx create-corvid-app <folder-name> [your-wix-site-url]
+    ```
+The [`create-corvid-app`](https://www.npmjs.com/package/create-corvid-app) would install `corvid-types` locally and you are ready to go.
```
Comment (wixysam) [inline README.md]: will install

File: README.md [hunk 11]
```diff
@@ -1,13 +1,38 @@
...
 
-# Publish
-update version in package.json
-run `npm publish` (git tag will be generated automaticly by `postpublish` script)
\ No newline at end of file
+### Manual Installation 
+1. Navigate to the folder where you have cloned your site to it and install the `corvid-types`
```
Comment (wixysam) [inline README.md]: your site ti and install `corvid-types`

File: README.md [hunk 12]
```diff
@@ -1,13 +1,38 @@
...
+### Manual Installation 
+1. Navigate to the folder where you have cloned your site to it and install the `corvid-types` 
+    ```
+    npm install --save-dev corvid-types
+    ```
+2. Open a Local Wix Editor:
+    ```
+    npx corvid open-editor
+    ```
+3. Hit the save button in the local editor window
```
Comment (wixysam) [inline README.md]: Click the **Save** button

File: CONTRIBUTING.md (hunk 2 above)
Comment (agankarin) [reply CONTRIBUTING.md]: Because it works better :)

**Pull request merged**